tests/unit/CoreSeparatorTest.php

```diff
@@ -0,0 +1,209 @@
+					}
+				}
+			}
+		';
+	}
+
+	/**
+	 * Test the retrieval of core/separator block fields and attributes.
+	 *
+	 * This test verifies that the block attributes for the core/separator block are
+	 * properly returned via the GraphQL query. The test ensures that the block content,
+	 * client ID, block name, and rendered HTML are correctly returned and that the attributes
+	 * (such as align, className, gradient, opacity, style and lock) match the expected values.
+	 *
+	 * @return void
+	 */
+	public function test_core_separator_fields_and_attributes() {
+		$block_content = '
+			<!-- wp:separator {"lock":{"move":true,"remove":true},"align":"wide","className":"is-style-dots","style":{"color":{"gradient":"linear-gradient(135deg,rgb(6,147,227) 1%,rgb(155,81,224) 100%)"}}} -->
+			<hr class="wp-block-separator alignwide has-alpha-channel-opacity"/>
+			<!-- /wp:separator -->
+		';
+
+		// Set post content.
+		wp_update_post(
+			[
+				'ID'           => $this->post_id,
+				'post_content' => $block_content,
+			]
+		);
+
+		$query     = $this->query();
+		$variables = [
+			'id' => $this->post_id,
+		];
+
+		$actual = graphql( compact( 'query', 'variables' ) );
+
+		$this->assertArrayNotHasKey( 'errors', $actual, 'There should not be any errors' );
+		$this->assertArrayHasKey( 'data', $actual, 'The data key should be present' );
+		$this->assertArrayHasKey( 'post', $actual['data'], 'The post key should be present' );
+		$this->assertEquals( $this->post_id, $actual['data']['post']['databaseId'], 'The post ID should match' );
+		$this->assertEquals( 1, count( $actual['data']['post']['editorBlocks'] ) );
+
+		$block = $actual['data']['post']['editorBlocks'][0];
+
+		// Verify the block data.
+		$this->assertNotEmpty( $block['apiVersion'], 'The apiVersion should be present' );
+		$this->assertEquals( 'design', $block['blockEditorCategoryName'], 'The blockEditorCategoryName should be text' );
+		$this->assertNotEmpty( $block['clientId'], 'The clientId should be present' );
+		$this->assertNotEmpty( $block['cssClassNames'], 'There should be cssClassNames' );
+		$this->assertEmpty( $block['innerBlocks'], 'There should be no inner blocks' );
+		$this->assertEquals( 'core/separator', $block['name'], 'The block name should be core/separator' );
+		$this->assertEmpty( $block['parentClientId'], 'There should be no parentClientId' );
+		$this->assertNotEmpty( $block['renderedHtml'], 'The renderedHtml should be present' );
+
+		$style = wp_json_encode(
+			[
+				'color' => [
+					'gradient' => 'linear-gradient(135deg,rgb(6,147,227) 1%,rgb(155,81,224) 100%)',
+				],
+			]
+		);
+
+		// Verify the attributes.
+		$this->assertEquals(
+			[
+				'align'           => 'wide',
+				'anchor'          => null,
+				'backgroundColor' => null,
+				'className'       => 'is-style-dots',
+				'cssClassName'    => 'wp-block-separator alignwide has-alpha-channel-opacity',
+				'gradient'        => null, // @todo: Getting null, but should be some valid gradient value.
```

Comment:
As noted, this is null because you're not using a gradient _preset_ ([ref](https://rtcamp.slack.com/archives/CK2CEH4F6/p1726759121200669?thread_ts=1726011552.525109&cid=CK2CEH4F6))


```suggestion
				'gradient'        => null,
```

Reply:
Fixed ✅